src/SmartComponents/CompletedTaskDetails/JobResultsDetails/EntryDetails.js

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+import propTypes from 'prop-types';
+import { Grid, GridItem, Label } from '@patternfly/react-core';
+import {
+  ExclamationCircleIcon,
+  ExclamationTriangleIcon,
+  InfoCircleIcon,
+} from '@patternfly/react-icons';
+
+const EntryRow = ({ entry }) => {
+  const { detail, key, severity, summary, title } = entry;
+
+  const renderLabel = (color, icon, text) => {
+    return (
+      <Label
+        color={color}
+        icon={icon}
+        style={{ marginTop: '16px', marginBottom: '4px' }}
+        className="pf-m-compact"
+      >
+        {text}
+      </Label>
+    );
+  };
```

Comment:
This works, and I know we do this all over the place in many apps, however, we should start to avoid functions like this. In React these functions will always have to be "rerendered" when the component does, this can affect rendering performance negatively. 

These functions look more like they should be a separate component, that way React does not need to rerender the function, but will be evaluated like a component and only rerender if it needs to.

My rule of thumb is, if it returns a component, it should probably be a component. There are exceptions, sometimes.